### PR TITLE
hap-py: init 0.3.15

### PR DESCRIPTION
**Pull request by apraga**
New package for genetics : helps compared variants called by a pipeline to a reference.

The only tests not passing are due to a difference in float representation (same value but with
different number of digits after decimal).

For best results (quality and running time), the
authors advice to use RTGtools vcfeval as the
engine. So this package requires  #230394

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by raspher** [comment]
Result of `nixpkgs-review pr 283881` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/by-name/ha/hap-py/package.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{ pkgs, stdenv, fetchFromGitHub, boost, zlib, lib, makeWrapper }:
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, boost, zlib, lib, makeWrapper }:
```
Using pkgs here is dirty and breaks cross

File: pkgs/by-name/ha/hap-py/package.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+let
+  # Bcftools needs perl
+  runtime = with pkgs; [ bcftools htslib my-python perl samtools ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:5]:
```suggestion
  runtime = [ bcftools htslib my-python perl samtools ];
```

File: pkgs/by-name/ha/hap-py/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  my-python-packages = p: with p; [
+    bx-python
+    pysam
+    cython
+    pandas
+    psutil
+    nose
+    scipy
+  ];
+  my-python = pkgs.python3.withPackages my-python-packages;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:15]:
```suggestion
  python = pkgs.python3.withPackages (p: with p; [
    bx-python
    pysam
    pandas
    psutil
    scipy
  ]);
```

nose is a test dependency usually and cython a compile time only dep

File: pkgs/by-name/ha/hap-py/package.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+in
+stdenv.mkDerivation rec {
+  pname = "hap.py";
+  version  = "0.3.15";
+  src = fetchFromGitHub {
+    owner = "Illumina";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-K8XXhioMGMHw56MKvp0Eo8S6R36JczBzGRaBz035zRQ=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:25]:
```suggestion
  pname = "hap.py";
  version  = "0.3.15";

  src = fetchFromGitHub {
    owner = "Illumina";
    repo = "hap.py";
    rev = "v${version}";
    sha256 = "sha256-K8XXhioMGMHw56MKvp0Eo8S6R36JczBzGRaBz035zRQ=";
  };

```

File: pkgs/by-name/ha/hap-py/package.nix [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  # For illumina script
+  BOOST_ROOT = "${boost.out}";
+  ZLIBSTATIC = "${zlib.static}";
+  # For cmake : boost lib and includedir are in differernt location
+  BOOST_LIBRARYDIR = "${boost.out}/lib";
+  BOOST_INCLUDEDIR = "${boost.dev}/include";
+  patches = [
+    ./hap-py.patch
+    ./cmake.patch
+  ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:35]:
```suggestion

  patches = [
    ./hap-py.patch
    ./cmake.patch
  ];

```
Why do we need those patches?

File: pkgs/by-name/ha/hap-py/package.nix [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  buildInputs = with pkgs; [
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:36]:
```suggestion
  buildInputs = [
```

File: pkgs/by-name/ha/hap-py/package.nix [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    autoconf
+    boost
+    bzip2
+    cmake
+    curl
+    htslib
+    my-python
+    rtg-tools
+    xz
+    zlib
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:46]: autoconf and cmake are definitely nativeBuildInputs.

File: pkgs/by-name/ha/hap-py/package.nix [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  ];
+  nativeBuildInputs = [ pkgs.makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:48]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/by-name/ha/hap-py/package.nix [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  propagatedBuildInputs = runtime;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:49]: Why propagate all things? Please use wrapper

File: pkgs/by-name/ha/hap-py/package.nix [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  postFixup = ''
+    wrapProgram $out/bin/hap.py \
+      --set PATH ${lib.makeBinPath runtime}
+       --add-flags "--engine-vcfeval-path=${rtg-tools}/bin/rtg
```
Comment (SuperSandro2000) [inline pkgs/by-name/ha/hap-py/package.nix:53]:
```suggestion
      --set PATH ${lib.makeBinPath runtime} \
      --add-flags "--engine-vcfeval-path=${rtg-tools}/bin/rtg
```

rtg-tools doesn't exist

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ha/hap-py/package.nix (hunk 10 above)
Comment (apraga) [reply pkgs/by-name/ha/hap-py/package.nix]: rtg-tools is the subject of a new PR that you also reviewed  #23039

**Comment by apraga** [comment]
Thanks for your helpful review ! The derivation has been improved to match your suggestions (except the one waiting on 
#23039).

**Comment by apraga** [comment]
PR needed for this PR has been merged (thanks !). If you are OK with this PR, could you merge it too ?

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**